internal/mode/static/state/graph/policies.go

```diff
@@ -0,0 +1,313 @@
+
+	if !ignored && ref.Nsname != client.ObjectKeyFromObject(gw.Source) {
+		return PolicyAncestor{}, false
+	}
+
+	ancestor = PolicyAncestor{
+		Ancestor: PolicyAncestorRef{
+			Kind:   "Gateway",
+			Group:  v1.GroupName,
+			Nsname: ref.Nsname,
+		},
+		Conditions: make([]conditions.Condition, 0),
+	}
+
+	if ignored {
+		ancestor.Conditions = append(ancestor.Conditions, staticConds.NewPolicyTargetNotFound("TargetRef is ignored"))
+
+		return ancestor, false
+	}
+
+	if !gw.Valid {
+		ancestor.Conditions = append(ancestor.Conditions, staticConds.NewPolicyTargetNotFound("TargetRef is invalid"))
+
+		return ancestor, false
+	}
+
+	if !policy.Valid {
+		return ancestor, false
+	}
+
+	ancestor.Conditions = append(ancestor.Conditions, staticConds.NewPolicyAccepted())
+
+	return ancestor, true
+}
+
+func processPolicies(
+	policies map[PolicyKey]policies.Policy,
+	validator validation.PolicyValidator,
+	gateways processedGateways,
+	routes map[types.NamespacedName]*Route,
+) map[PolicyKey]*Policy {
+	if len(policies) == 0 || gateways.Winner == nil {
+		return nil
+	}
+
+	processedPolicies := make(map[PolicyKey]*Policy)
+
+	for key, policy := range policies {
+		ref := policy.GetTargetRef()
+		refNsName := targetRefNsName(ref, policy.GetNamespace())
+
+		refGroupKind := fmt.Sprintf("%s/%s", ref.Group, ref.Kind)
+
+		switch refGroupKind {
+		case gatewayGroupKind:
+			if !gatewayExists(refNsName, gateways.Winner, gateways.Ignored) {
+				continue
+			}
+		case hrGroupKind:
+			if _, exists := routes[refNsName]; !exists {
+				continue
+			}
+		}
+
+		conds := make([]conditions.Condition, 0, 2)
+
+		if err := validator.Validate(policy); err != nil {
+			conds = append(conds, staticConds.NewPolicyInvalid(err.Error()))
+		}
+
+		processedPolicies[key] = &Policy{
+			Source:     policy,
+			Valid:      len(conds) == 0,
+			Conditions: conds,
+			TargetRef: PolicyTargetRef{
+				Kind:   ref.Kind,
+				Group:  ref.Group,
+				Nsname: refNsName,
+			},
+		}
+	}
+
+	markConflictedPolicies(processedPolicies, validator)
+
+	return processedPolicies
+}
+
+// markConflictedPolicies marks policies that conflict with a policy of greater precedence as invalid.
+// Policies are sorted by timestamp and then alphabetically.
+func markConflictedPolicies(policies map[PolicyKey]*Policy, validator validation.PolicyValidator) {
+	type key struct {
+		policyGVK schema.GroupVersionKind
+		PolicyTargetRef
+	}
+
+	possibles := make(map[key][]*Policy)
+
+	for pk, p := range policies {
+		if p.Valid {
+			ak := key{
+				PolicyTargetRef: p.TargetRef,
+				policyGVK:       pk.GVK,
+			}
+			if possibles[ak] == nil {
+				possibles[ak] = make([]*Policy, 0)
+			}
+			possibles[ak] = append(possibles[ak], p)
+		}
+	}
+
+	for _, policyList := range possibles {
+		if len(policyList) > 1 {
+			sort.SliceStable(
+				policyList, func(i, j int) bool {
+					return ngfsort.ClientObject(policyList[i].Source, policyList[j].Source)
+				},
+			)
+
+			for i := range policyList {
```

Comment:
this part is a bit complicated (imho)
it looks like every policy could conflict with every other policy.
However, once we iterate over loop, we `for j := i + 1...` , we mark some conflicted policies as invalid, which I think combined with 
```
				if !policyList[i].Valid {
					continue
				}
```
might free some other policies later in the sorted order from conflicts...

I wonder if it is possible to add a comment that clarifies the logic